sympy/simplify/combsimp.py

```diff
@@ -64,450 +47,82 @@ def combsimp(expr):
+        else:
+            return rv
 
-        # =========== level 2 work: pure gamma manipulation =========
+        hit = False
+        for i in range(len(d_args)):
+            ai = d_args[i]
+            if not isinstance(ai, factorial):
+                continue
 
-        # Try to reduce the number of gamma factors by applying the
-        # reflection formula gamma(x)*gamma(1-x) = pi/sin(pi*x)
-        for gammas, numer, denom in [(
-            numer_gammas, numer_others, denom_others),
-                (denom_gammas, denom_others, numer_others)]:
-            new = []
-            while gammas:
-                g1 = gammas.pop()
-                if g1.is_integer:
-                    new.append(g1)
+            for j in range(i + 1, len(d_args)):
+                aj = d_args[j]
+                if not isinstance(aj, factorial):
                     continue
-                for i, g2 in enumerate(gammas):
-                    n = g1 + g2 - 1
-                    if not n.is_Integer:
-                        continue
-                    numer.append(S.Pi)
-                    denom.append(sin(S.Pi*g1))
-                    gammas.pop(i)
-                    if n > 0:
-                        for k in range(n):
-                            numer.append(1 - g1 + k)
-                    elif n < 0:
-                        for k in range(-n):
-                            denom.append(-g1 - k)
-                    break
-                else:
-                    new.append(g1)
-            # /!\ updating IN PLACE
-            gammas[:] = new
 
-        # Try to reduce the number of gammas by using the duplication
-        # theorem to cancel an upper and lower: gamma(2*s)/gamma(s) =
-        # 2**(2*s + 1)/(4*sqrt(pi))*gamma(s + 1/2). Although this could
-        # be done with higher argument ratios like gamma(3*x)/gamma(x),
-        # this would not reduce the number of gammas as in this case.
-        for ng, dg, no, do in [(numer_gammas, denom_gammas, numer_others,
-                                denom_others),
-                               (denom_gammas, numer_gammas, denom_others,
-                                numer_others)]:
+                sum = ai.args[0] + aj.args[0]
+                if sum in n_fact_args:
+                    n_args.remove(factorial(sum))
+                    n_fact_args.remove(sum)
+                    n_args.append(binomial(sum, ai.args[0] if count_ops(
+                    ai.args[0]) < count_ops(aj.args[0]) else aj.args[0]))
 
-            while True:
-                for x in ng:
-                    for y in dg:
-                        n = x - 2*y
-                        if n.is_Integer:
-                            break
-                    else:
-                        continue
-                    break
-                else:
+                    d_args[i] = None
+                    d_args[j] = None
+                    hit = True
                     break
```

Comment:
Could there be another loop where the roles of `n` and `d` have been interchanged?